ui-tests/cypress/integration/administration/email-configuration-Sender.spec.js

```diff
@@ -0,0 +1,306 @@
+
+      });
+
+      it([Tag.SANITY, 'ENG-3299'], 'Update an existing sender', () => {
+
+        currentPage.getContent().getActionButton().click();
+        currentPage.getContent().getEditButton().click().wait(1000);
+        cy.validateAppBuilderUrlPathname('/email-config/senders/edit/CODE1');
+        currentPage.getContent()
+                   .getEmailInput()
+                   .clear()
+                   .type(testEmail);
+        currentPage.getContent().senderSubmit()
+                   .click().wait(500);
+        currentPage.getContent()
+                   .getSenderTable()
+                   .contains(testEmail)
+                   .should('be.visible');
+        cy.validateToast(currentPage);
+        cy.senderController()
+          .then(controller => controller.defaultEditSender(defaultCode1)).wait(1000);
+
+      });
+
+      it([Tag.SANITY, 'ENG-3299'], 'Delete Modal is displayed', () => {
+
+
+        currentPage.getContent().getActionButton().click();
+        currentPage.getContent().getDeleteButton().click().wait(1000);
+        currentPage.getDialog()
+                   .get().children(`${htmlElements.div}#DeleteSenderModal`)
+                   .should('be.visible');
+
+       });
+
+      it([Tag.SANITY, 'ENG-3299'], 'Delete a sender', () => {
+
+
+        currentPage.getContent().getActionButton().click();
+        currentPage.getContent().getDeleteButton().click().wait(1000);
+        currentPage.getDialog().confirm();
+        cy.validateToast(currentPage);
+        currentPage.getContent()
+                   .getSenderTable()
+                   .should('not.contain', 'CODE1');
+        cy.senderController()
+                   .then(controller => controller.addDefaultSender(defaultCode1, defaultEmail1)).wait(1000);
+
+
+  });
+      it([Tag.SANITY, 'ENG-3299'], 'Click on Cancel Modal Button ', () => {
+
+        currentPage.getContent().getActionButton().click();
+        currentPage.getContent().getDeleteButton().click().wait(1000);
+        currentPage.getDialog().getCancelButton().click();
+        currentPage.getContent().getSenderTable().should('contain', 'CODE1');
+
+      });
+
+    });
+
+    describe([Tag.FEATURE, 'ENG-3299'], 'Feature Test', () => {
+       
+      
+      it([Tag.FEATURE, 'ENG-3299'], 'No Sender Exist ', () => {
+
+       
+        cy.senderController()
+        .then(controller => controller.deleteSender(defaultCode1)).wait(500);
+        cy.senderController()
+        .then(controller => controller.deleteSender(defaultCode2)).wait(500);
+        currentPage = openSenderPage();
+        currentPage.getContent()
+                    .getSenderTable()
+                    .should('be.visible');
+        currentPage.getContent()
+                    .getSenderTable()
+                    .should('not.contain', 'CODE1');
+        currentPage.getContent()
+                    .getSenderTable()
+                    .should('not.contain', 'CODE2');
```

Comment:
Is this enough to ensure the list is empty? It might contain other elements and this test would still pass

Reply:
I added a "tbody should not be visible".